src/util/Whamm.v3

```diff
@@ -43,24 +43,40 @@ type WhammArg {
 	case Local(t: ValueType, i: int);
 }
 
+type WhammPredicate {
+	case True;
+	case WhammParam(call: WhammParam);
+}
+
 // Utilities associated with Whamm probes.
 component Whamm {
 	def NO_PARAMS: Array<WhammParam> = [];
 
 	// Parse a string representing an opcode match with optional parameters.
-	def parseOpcodePattern(r: TextReader) -> (Opcode, Array<WhammParam>) {
+	def parseOpcodePattern(r: TextReader) -> (Opcode, Array<WhammParam>, WhammPredicate) {
 		if (r.optN("wasm:") >= 0) {
 			if (r.optN("opcode:") >= 0) {
 				var op = Opcodes.parseName(r.data[r.pos ...]);
 				if (op != Opcode.INVALID) {
 					r.advance(op.mnemonic.length);
+
+					var pred: WhammParam;
+					r.skipWhitespace();
+					if (r.char == '/') {
+						r.req1('/');
+						r.skipWhitespace();
+						pred = parseParam0(r);
+						r.skipWhitespace();
```

Comment:
I think it'd be good to enforce the syntactic restriction that the predicate is a function call here, and make other cases unrepresentable.